contracts/TokenBurnDrop.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.4.24;
+
+import "./DSAuth.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./interfaces/ISettingsRegistry.sol";
+import "./interfaces/ERC223.sol";
+import "./SettingIds.sol";
+
+contract TokenBurnDrop is DSAuth, SettingIds {
+    // claimedToken event
+    event ClaimedTokens(address indexed token, address indexed owner, uint amount);
+
+    // burndropTokens event
+    event BurndropTokens(address indexed token, address indexed owner, uint amount, bytes data);
+
+    ISettingsRegistry public registry;
+
+    function initializeContract(address _registry) public onlyOwner{
+        registry = ISettingsRegistry(_registry);
+    }
+
+    /**
+    * @dev ERC223 fallback function, make sure to check the msg.sender is from target token contracts
+    * @param _from - person who transfer token in for deposits or claim deposit with penalty KTON.
+    * @param _amount - amount of token.
+    * @param _data - data which indicate the operations.
+    */
+    function tokenFallback(address _from, uint256 _amount, bytes _data) public {
+        address ring = registry.addressOf(SettingIds.CONTRACT_RING_ERC20_TOKEN);
+        address kryptonite = registry.addressOf(SettingIds.CONTRACT_KTON_ERC20_TOKEN);
+
+        require((msg.sender == ring) || (msg.sender == kryptonite), "Permission denied");
+
+        require(_data.length == 32, "The address (Darwinia Network) must be in a 64-bit hexadecimal format");
+
+        //  burndrop ring
+        if(ring == msg.sender) {
+            ERC223(ring).transfer(address(0), _amount);
+        }
+
+        //  burndrop kton
+        if (kryptonite == msg.sender) {
+            ERC223(kryptonite).transfer(address(0), _amount);
+        }
+
+        emit BurndropTokens(msg.sender, _from, _amount, _data);
```

Comment:
If the msg.sender if not from RING or KTON, this event will also be triggers, which will the verifiers' effort to filter and check the sender.

Better to filter in the contract here.

Reply:
```suggestion
        require((msg.sender == ring) || (msg.sender == kryptonite), "Permission denied");
        emit BurndropTokens(msg.sender, _from, _amount, _data);
```